File: ALODWebUtility/Common/SARCUtility.vb
Imports ALOD.Core.Domain.Common
Imports ALOD.Core.Domain.Modules.SARC
Imports ALOD.Data.Services

Namespace Common
    Public Module SARCUtility

        Public Function CreateSARCFinding(ByVal sarcId As Integer) As RestrictedSARCFindings
            Dim cFinding As RestrictedSARCFindings = New RestrictedSARCFindings()
            Dim currUser = UserService.CurrentUser

            cFinding.SARCID = sarcId
            cFinding.SSN = currUser.SSN
            cFinding.Compo = currUser.Component
            cFinding.Rank = currUser.Rank.Rank
            cFinding.Grade = currUser.Rank.Grade
            cFinding.Name = currUser.FullName
            cFinding.ModifiedBy = currUser.Id
            cFinding.ModifiedDate = DateTime.Now
            cFinding.CreatedBy = currUser.Id
            cFinding.CreatedDate = DateTime.Now

            Return cFinding
        End Function

        Public Function HasFinding(ByVal finding As RestrictedSARCFindings) As Boolean
            If (finding Is Nothing) Then
                Return False
            End If

            If (Not finding.Finding.HasValue OrElse finding.Finding.Value = 0) Then
                Return False
            End If

            Return True
        End Function

        Public Function IsTimeToSaveViaNavigator(ByVal buttonType As NavigatorButtonType) As Boolean
            If (buttonType = NavigatorButtonType.Save OrElse
                buttonType = NavigatorButtonType.NavigatedAway OrElse
                buttonType = NavigatorButtonType.NextStep OrElse
                buttonType = NavigatorButtonType.PreviousStep) Then
                Return True
            End If

            Return False
        End Function

    End Module
End Namespace

File: ALODWebUtility.CS/Common/SARCUtility.cs
using System;
using ALOD.Core.Domain.Common;
using ALOD.Core.Domain.Modules.SARC;
using ALOD.Data.Services;

namespace ALODWebUtility.Common
{
    public static class SARCUtility
    {
        public static RestrictedSARCFindings CreateSARCFinding(int sarcId)
        {
            RestrictedSARCFindings cFinding = new RestrictedSARCFindings();
            var currUser = UserService.CurrentUser();

            cFinding.SARCID = sarcId;
            cFinding.SSN = currUser.SSN;
            cFinding.Compo = currUser.Component;
            cFinding.Rank = currUser.Rank.Rank;
            cFinding.Grade = currUser.Rank.Grade;
            cFinding.Name = currUser.FullName;
            cFinding.ModifiedBy = currUser.Id;
            cFinding.ModifiedDate = DateTime.Now;
            cFinding.CreatedBy = currUser.Id;
            cFinding.CreatedDate = DateTime.Now;

            return cFinding;
        }

        public static bool HasFinding(RestrictedSARCFindings finding)
        {
            if (finding == null)
            {
                return false;
            }

            if (!finding.Finding.HasValue || finding.Finding.Value == 0)
            {
                return false;
            }

            return true;
        }

        public static bool IsTimeToSaveViaNavigator(NavigatorButtonType buttonType)
        {
            if (buttonType == NavigatorButtonType.Save ||
                buttonType == NavigatorButtonType.NavigatedAway ||
                buttonType == NavigatorButtonType.NextStep ||
                buttonType == NavigatorButtonType.PreviousStep)
            {
                return true;
            }

            return false;
        }
    }
}
